examples/development/concurrency/isolates/lib/main.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:isolate';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Isolates demo',
+      home: Scaffold(
+        appBar: AppBar(
+          title: const Text('Isolates demo'),
+        ),
+        body: Center(
+          child: ElevatedButton(
+            onPressed: () {
+              getPhotos();
+            },
+            child: Text('Fetch photos'),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+// #docregion isolate-run
+// Produces a list of 211,640 photo objects.
+// (The JSON file is ~20MB.)
+Future<List<Photo>> getPhotos() async {
```

Comment:
Can we improve the types here a bit? Primarily giving `Isolate.run()` its type argument,
and using a cast<T>() for the List, which makes it a bit cleaner. I do not feel strongly about changing dynamic -> `Object?`

```suggestion
Future<List<Photo>> getPhotos() async {
  final String jsonString = await rootBundle.loadString('assets/photos.json');

  final List<Photo> photos = await Isolate.run<List<Photo>>(() {
    // jsonDecode() gives untyped lists.
    final List<Object?> photoData = jsonDecode(jsonString) as List<Object?>;
    // So cast here, and then you could use a tear-off as well.
    return photoData.cast<Map<String, Object?>>().map(Photo.fromJson).toList();
  });

  return photos;
}
```

Reply:
I'm adding this suggested update to my next commit, as I'll need to update the snippets in the markdown docs that this uses as well.